kr8s/_api.py

```diff
@@ -130,7 +131,7 @@ def _construct_url(
         parts = [base]
         if version:
             parts.append(version)
-        if namespace is not None:
+        if namespace:
```

Comment:
When using a URL with a base url component and setting `namespace=kr8s.ALL` this can end up as an empty string. So modifying the if statement to catch that case.